com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StartWithoutDebuggingRequestsHandler.java

```diff
@@ -0,0 +1,168 @@
+        } else if (Command.DISCONNECT.equals(command)) {
+            return handleDisconnectCommand(arguments, response, context);
+        } else {
+            if (context.getLaunchMode() == LaunchMode.NO_DEBUG) {
+                String errorMsg = String.format("Unrecognized request: { _request: %s }", command);
+                throw AdapterUtils.createCompletionException(errorMsg, ErrorCode.UNRECOGNIZED_REQUEST_FAILURE);
+            }
+            return CompletableFuture.completedFuture(response);
+        }
+    }
+
+    @Override
+    protected Process launchInternalDebuggeeProcess(LaunchArguments launchArguments, IDebugAdapterContext context)
+            throws IOException, IllegalConnectorArgumentsException, VMStartException {
+        String[] cmds = constructLaunchCommands(launchArguments, false, null);
+        File workingDir = null;
+        if (launchArguments.cwd != null && Files.isDirectory(Paths.get(launchArguments.cwd))) {
+            workingDir = new File(launchArguments.cwd);
+        }
+        this.debuggeeProcess = Runtime.getRuntime().exec(cmds, constructEnvironmentVaraiables(launchArguments),
+                workingDir);
+        new Thread() {
+            public void run() {
+                try {
+                    debuggeeProcess.waitFor();
+                } catch (InterruptedException ignore) {
+                    logger.warning(String.format("Current thread is interrupted. Reason: %s", ignore.toString()));
+                    debuggeeProcess.destroy();
+                } finally {
+                    context.getProtocolServer().sendEvent(new Events.TerminatedEvent());
+                }
+            }
+        }.start();
+        logger.info("Launching debuggee proccess succeeded.");
+        return this.debuggeeProcess;
+    }
+
+    @Override
+    protected void postLaunchConfiguration(LaunchArguments launchArguments, IDebugAdapterContext context) {
+        // For NO_DEBUG launch mode, the debugger does not respond to requests like
+        // SetBreakpointsRequest,
+        // but the front end keeps sending them according to the Debug Adapter Protocol.
+        // To avoid receiving them, a workaround is not to send InitializedEvent back to
+        // the front end.
+        // See https://github.com/Microsoft/vscode/issues/55850#issuecomment-412819676
+        return;
+    }
+
+    @Override
+    protected CompletableFuture<Response> launchInTerminal(LaunchArguments launchArguments, Response response,
+            IDebugAdapterContext context) {
+        CompletableFuture<Response> resultFuture = new CompletableFuture<>();
+
+        final String launchInTerminalErrorFormat = "Failed to launch debuggee in terminal. Reason: %s";
+
+        String[] cmds = constructLaunchCommands(launchArguments, false, null);
+        RunInTerminalRequestArguments requestArgs = null;
```

Comment:
The logic to construct launchInTerminal request arguments and handle the response is similar as the normal LaunchRequestHandler, suggest to extract it to a helper method in the superclass.

Reply:
Agree. Suggest to do that in another new PR.